### DISCUSSION TITLE
Split Bar but extend to notch

### DISCUSSION THREAD

**Discussion by neutonfoo** [FelixKratz/SketchyBar#292] [Q&A] [answered]
I have a MacBook with a notch and I was wondering if it would be possible to have two full bars that extend from the top left corner of the screen to the left side of the notch AND from the right side of the notch to the top right corner of the screen. I've tried using brackets with two items placed in `left` and `q` but they don't seem to extend properly.

<details>
  <summary>sketchybarrc</summary>
  
```sh
#!/usr/bin/env zsh

PLUGIN_DIR="$HOME/.config/sketchybar/plugins"

##### Bar Appearance #####
# https://felixkratz.github.io/SketchyBar/config/bar

sketchybar	--bar	height=32 \
                    blur_radius=50 \
                    position=top \
                    sticky=on \
                    padding_left=23 \
                    padding_right=23 \
                    display=main \
                    color=0x00000000

sketchybar	--add	item	test1 left \
            --set	test1 \
                    	icon=1

sketchybar	--add	item	test2 q \
            --set	test2 \
                    	icon=2

sketchybar	--add	bracket left_bar test1 test2 \
            --set	left_bar \
                        background.color=0xff000000

##### Adding Left Items #####
# https://felixkratz.github.io/SketchyBar/config/items#adding-items-to-sketchybar
##### Finalizing Setup #####
sketchybar --update

echo "sketchybar configuration loaded.."
```
</details>

Screenshot

<img width="1470" alt="Screenshot 2022-11-17 at 9 57 11 PM" src="https://user-images.githubusercontent.com/3114585/202465504-f3dd719b-d81a-4065-94d1-2b25c7141ed7.png">

**Comment by FelixKratz** [comment]
I think this is not really achievable with brackets. One way this could be done however is with  a background `clip` and a regular fully connected bar:
```bash
sketchybar --add item split center \
           --set split width=200 \
                       background.color=0x00000000 \
                       background.clip=1.0 \
                       background.height=50 \
                       background.corner_radius=0

```
This works since the background of this center item clips the bars background and makes it transparent.

This will look strange when the bar has a `shadow` or `corner_radius`, but will do the job perfectly for a square bar with no shadow:
![Screenshot 2022-11-17 at 15 23 50](https://user-images.githubusercontent.com/22680421/202471764-2d5010fb-0147-43b0-ae61-1777a2b0e7d5.png)

`sketchybarrc`:
```bash
sketchybar --bar margin=10 \
                 y_offset=10 \
                 color=0xff000000 \
                 height=40

sketchybar --add item split center \
           --set split width=200 \
                       background.color=0x00000000 \
                       background.clip=1.0 \
                       background.height=50 \
                       background.corner_radius=0

```

**Reply by neutonfoo** [reply]
Thanks for the answer! I was actually hoping to get a corner radius on it so it would be like 2 pills around my notch haha, but this is close enough.

**Reply by FelixKratz** [reply]
If you can accept a good amount of sketchyness it can also be done with rounded borders (by overlaying the bar with a popup on either side of the gap):
![Screenshot 2022-11-17 at 15 46 00](https://user-images.githubusercontent.com/22680421/202477642-37827727-f19a-407a-a68f-e379f51e20ee.png)

`sketchybarrc`:
```bash
COLOR=0xff000000
HEIGHT=40

sketchybar --bar margin=10 \
                 y_offset=10 \
                 color=$COLOR \
                 corner_radius=11 \
                 topmost=on \
                 height=$HEIGHT

sketchybar --add item split.left center \
           --set split.left icon.drawing=off \
                             label.drawing=off \
                             popup.drawing=on \
                             popup.y_offset=-$HEIGHT \
                             popup.background.color=$COLOR \
                             popup.background.corner_radius=11 \
                             popup.align=center \
                             popup.height=40

sketchybar --add item spacer.left popup.split.left \
           --set spacer.left width=22

sketchybar --add item split center \
           --set split width=200 \
                       background.color=0x00000000 \
                       background.clip=1.0 \
                       background.height=$HEIGHT \
                       background.corner_radius=0

sketchybar --add item split.right center \
           --set split.right icon.drawing=off \
                             label.drawing=off \
                             popup.drawing=on \
                             popup.y_offset=-$HEIGHT \
                             popup.background.color=$COLOR \
                             popup.background.corner_radius=11 \
                             popup.align=center \
                             popup.height=40

sketchybar --add item spacer.right popup.split.right \
           --set spacer.right width=22
```

 This will look strange on a secondary display.


None of this is ideal, I will think about a nicer solution for this.

**Reply by neutonfoo** [reply]
Hah wow this is truly amazing. Thanks for taking the time to write this.

**Reply by FelixKratz** [reply]
I have worked on the bracket system and on current master it is possible to create brackets between any items, no matter where they are positioned in the bar and the bounds will be calculated correctly automatically. I.e. on current master you can achieve this design via brackets easily if you bracket first `left` item and first `e` item. Analogously with `right` and `q`.

**Reply by neutonfoo** [reply]
Amazing!! Time to rework my config. Thanks for working so hard on this. Edit: I don't build sketchybar, so my version is always whatever's on Homebrew. Any idea when main will be pushed out to Homebrew?

**Reply by FelixKratz** [reply]
> I don't build sketchybar, so my version is always whatever's on Homebrew. Any idea when main will be pushed out to Homebrew?

I am testing it for a few days to hopefully spot all the quirks and then push it to brew on the weekend.

**Comment by FelixKratz** [comment, marked as answer]
With the newest release bracketing the first `left` and the first `q` item is the way to go. I will hide the old hacky "solutions" as deprecated.
